velox/exec/HashProbe.cpp

```diff
@@ -1212,11 +1212,12 @@ void HashProbe::prepareFilterRowsForNullAwareJoin(
     for (auto& projection : filterInputProjections_) {
       filterInputColumnDecodedVector_.decode(
           *filterInput->childAt(projection.outputChannel), filterInputRows_);
-      if (filterInputColumnDecodedVector_.mayHaveNulls()) {
+      if (const uint64_t* nulls =
```

Comment:
nit: `if (filterInputColumnDecodedVector_.mayHaveNulls() && const uint64_t* nulls = filterInputColumnDecodedVector_.nulls(&filterInputRows_)) ...`

mayHaveNulls() is a cheap function while nulls(&filterInputRows_) is expensive. We wouldn't want to do the expensive one unless filterInputColumnDecodedVector_ may have nulls.

Reply:
That's right! I don't think that's valid syntax though. I tweaked it a little so it is valid. Thanks Wei!